app/queries/check_for_content_item_preventing_draft_from_being_published.rb

```diff
@@ -0,0 +1,53 @@
+module Queries
+  module CheckForContentItemPreventingDraftFromBeingPublished
+    extend ArelHelpers
+
+    # Checks for any content item which would prevent a content item with the
+    # specified content_id and base_path from being published (from the draft
+    # state).
+    def self.call(content_id, base_path, document_type)
+      if SubstitutionHelper::SUBSTITUTABLE_DOCUMENT_TYPES.include?(document_type)
+        nil # The SubstitionHelper will unpublish any item that is in the way
```

Comment:
Do you mean what, or why?

Assuming you mean why, and not what, the publish command invokes the substitution helper, which will unpublish a "blocking" item if either the item being published, or the blocking item have a substitutable document type. This query is trying to work out if something can be published by detecting blocking items. To deal with the case where the substitution helper would be involved due to the blocking items document type, the query excludes content items with a substitutable document type. To deal with the case where the item being published has a substitutable document type, if the document type specified is substitutable, there is no point looking for blocking items, as if one does exist, it will be handled by the substitution helper. This if is that last check.

If this makes sense, I could add more of this information in to the code?